editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -903,6 +903,69 @@ impl<'a> MessageHandler<NodeGraphMessage, NodeGraphHandlerData<'a>> for NodeGrap
 						};
 						responses.add(FrontendMessage::UpdateWirePathInProgress { wire_path: Some(wire_path) });
 					}
+				} else if self.disconnecting.is_some() {
+					// Disconnecting with no upstream node, create new value node.
+					let to_connector = network_interface.input_connector_from_click(ipp.mouse.position, selection_network_path);
+					if let Some(to_connector) = &to_connector {
+						let Some(input_position) = network_interface.input_position(to_connector, selection_network_path) else {
+							log::error!("Could not get input position for connector: {to_connector:?}");
+							return;
+						};
+						self.wire_in_progress_to_connector = Some(input_position);
+					}
+					// Not hovering over a node input or node output, insert the node
+					else {
+						// Disconnect if the wire was previously connected to an input
+						if let Some(disconnecting) = self.disconnecting.take() {
+							let mut position = if let Some(to_connector) = self.wire_in_progress_to_connector { to_connector } else { point };
+							// Offset to drag from center of node
+							position = position - DVec2::new(24. * 3., 24.);
```

Comment:
Magic 24 and 3? Perhaps add some named constants since it isn't really clear what these represent.